lms/djangoapps/grades/new/course_grade_factory.py

```diff
@@ -123,6 +123,8 @@ def iter(
                     )
                     yield self.GradeResult(user, None, exc)
 
+        VisibleBlocks.clear_cache(course_data.course_key)
```

Comment:
Nitty: This line seems a bit out of place.  It may read better if the code was instead surrounded by a VisibleBlocks transaction, such as:
```
VisibleBlocks.begin_course_transaction(course_data.course_key)
# do a bunch of stuff using Visible Blocks for the course
VisibleBlocks.end_course_transaction(course_data.course_key)
```

`begin_course_transaction` would be a no-op.
`end_course_transaction` would call `clear_cache`.